lib/cli/src/commands/check/utils/transform-spec-composition.ts

```diff
@@ -0,0 +1,429 @@
+/**
+ * OpenAPI spec transformation utilities for CommonGrants CLI validation.
+ *
+ * PROBLEM
+ * -------
+ * The CommonGrants CLI validation tool expects OpenAPI specs to follow specific composition patterns that match the CommonGrants protocol base specification. However, different web frameworks generate OpenAPI specs with different composition patterns that cause validation to fail.
+ *
+ * For example, APIFlask generates schemas like this:
+ * ```yaml
+ * paginationInfo:
+ *   type: [object]          # ← Problem: unnecessary type field
+ *   allOf:
+ *   - $ref: '#/components/schemas/PaginatedResultsInfo'
+ * ```
+ *
+ * But the CommonGrants protocol base spec uses:
+ * ```yaml
+ * paginationInfo:
+ *   allOf:                  # ← Expected: just allOf, no type field
+ *   - $ref: '#/components/schemas/CommonGrants.Pagination.PaginatedResultsInfo'
+ * ```
+ *
+ * This causes validation errors like:
+ * "Type mismatch. Base is 'object', impl is 'object'"
+ *
+ * SOLUTION
+ * --------
+ * Transform the implementation spec to match the base spec's composition patterns before validation. This is done by:
+ *
+ * 1. **Schema Inlining**: Replace `$ref` references in response schemas with the actual schema definitions to enable composition pattern fixes.
+ *
+ * 2. **Composition Pattern Fixing**: Recursively find and fix the problematic pattern:
+ *    - Find: `type: [object]` + `allOf` (or `type: "object"` + `allOf`)
+ *    - Fix: Remove the `type` field, keep only `allOf`
+ *
+ * 3. **Structure Normalization**: Convert `allOf` patterns with single `$ref` items to direct `$ref` patterns to match the base spec structure.
```

Comment:
I'm not sure this is necessary, because the CommonGrants base spec has plenty of the `allOf` with a single `$ref`:

```yaml
schema:
  type: object
  required:
    - data
  properties:
    data:
      allOf:
        - $ref: '#/components/schemas/CommonGrants.Models.AppFormResponse'
      description: Response data
  allOf:
    - $ref: '#/components/schemas/CommonGrants.Responses.Success'
```